### PR TITLE
Finish modeling-users

### PR DESCRIPTION
**Pull request by quoctv1202**
Chap 6: Finish modeling-users.

### CODE REVIEW

**Comment by quoctv1202** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  before_save email_downcase
```
Comment (huuphuc93) [inline app/models/user.rb]: :email_downcase

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  validates :name,  presence: true, length: {maximum: Settings.users.name_length}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: Settings.users.email_length},
+    format: {with: VALID_EMAIL_REGEX},uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: { minimum: Settings.use.pass_length_min }
```
Comment (huuphuc93) [inline app/models/user.rb]: {minimum: Settings.use.pass_length_min}

**Comment by quoctv1202** [comment]
ready

File: app/models/user.rb (hunk 2 above)
Comment (quoctv1202) [reply app/models/user.rb]: sửa xong 2 cái này e merge được chưa a

**Comment by quoctv1202** [comment]
ready

**Comment by quoctv1202** [comment]
ready

**Comment by quoctv1202** [comment]
ready.

**Comment by quoctv1202** [comment]
ready

**Comment by huuphuc93** [comment]
merged

**Pull request merged**